### ISSUE TITLE
Virtualising AArch64 with UEFI - virt-manager complains about insufficient block storage

### ISSUE THREAD

**Issue by frantisekhanzlikbl** [NixOS/nixpkgs#175202]
### Describe the bug

I tried to create a virtual machine using virt-manager, but before starting the installation, virt-manager throws the error:
```
Unable to complete install: 'internal error: qemu unexpectedly closed the monitor: 2022-05-28T20:56:21.969616Z qemu-system-aarch64: device requires 67108864 bytes, block backend provides 3653632 bytes'
```

### Steps To Reproduce
Steps to reproduce the behaviour:
1. Inside virt-manager, create a new virtual machine.
2. Select the `aarch64` architecture.

   Note: after doing this, virt-manager disables all the installation source options except importing existing disk image and manual install, and shows a warning: `Warning: Failed to setup UEFI: Did not find any UEFI binary path for arch 'aarch64'`
3. select nixos-unstable as the installed system. (I am unsure if this is relevant)
4. Leave the memory & CPU options at their defaults (3072 MiB memory, 1 CPU)
5. Select the 'customize configuration before install' checkbox.
6. Change the firmware to `UEFI x86_64: /run/libvirt/nix-ovmf/OVMF_CODE.fd`

   Note: The x86_64 here seems suspicious to me, but I'm not sure if it's a problem or not.
8. Begin the installation
9. The aforementioned error pops up.

### Expected behavior
I am allowed to continue installing the OS.

### Screenshots

The warning after selecting the aarch64 arch:
![image](https://user-images.githubusercontent.com/43645405/170843053-02cbce7e-308c-43a6-ae2a-eeb9f380c5e5.png)

The configuration dialogue:
![image](https://user-images.githubusercontent.com/43645405/170843000-e8d233f3-d533-43bb-b648-23bbf16ae15a.png)

### Additional context

Please note that my understanding of virtualisation and related topics is extremely limited, so while I'd love to try to find out anything that I can, I do not have any idea about how to even begin diagnosing or solving the problem.

Probably expected, but the size that it is complaining about being too low (3653632) is exactly equal to the size of `/run/libvirt/nix-ovmf/OVMF_CODE.fd`.

### Notify maintainers
I am unsure who to ping about this, but my guess would be:
Maintainers of `libvirt`: @fpletz @globin @lovesegfault 

### Metadata

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.39, NixOS, 22.05 (Quokka)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.8.1`
 - channels(root): `"home-manager, nixos-21.11.336020.2128d0aa28e, nixos-unstable-22.05pre356180.d5f23787297"`
 - channels(frantisek_hanzlik): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by frantisekhanzlikbl** [comment]
After some more research, the OVMF arch indeed seems to be the cause, so now I am trying to obtain OVMF for aarch64.

I still don't know what exactly does the file contain, but it seems like it should not depend on anything that would cause problems when using an OVMF file from another distro. I'd prefer to use the one provided by NixOS, but so far I have been unsuccessful at cross-compiling the `OVMF` package because it depends on `edf2`, which seems to have broken dependencies for cross-compilation.

**Comment by frantisekhanzlikbl** [comment]
I have eventually got `edf2` building, but I gave up on fixing `OVMF`.
To my happiness, when I downloaded AAVMF from Fedora's repos (https://fedora.pkgs.org/36/fedora-updates-aarch64/edk2-aarch64-20220221gitb24306f15daa-4.fc36.noarch.rpm.html), everything works as expected.

I am still interested in cross-compiling OVMF with nix, but I'll open a new issue for that.

**Comment by frantisekhanzlikbl** [comment]
@twz123 Sorry, I was swamped with work and forgot to create it :sweat_smile:
It's pretty late in here now, but I'll try to get to it tomorrow

**Comment by frantisekhanzlikbl** [comment]
Oh, the wonders of OSS. Half a year later, I've come to need to virtualise UEFI aarch64 again, and this time, OVMF has cross-compiled without any issues and works great, even when overriding stuff.

Not sure whether that's thanks to anyone mentioned here, but I'm really grateful for the nix community. Everything Nix is basically a dream-come-true, and it's great seeing even such obscure issues being solved!

**Comment by fl42v** [comment]
Just in case anyone else has the same problem, make sure to have

```Nix
virtualisation.libvirtd.qemu.ovmf.packages = [
  pkgs.pkgsCross.aarch64-multiplatform.OVMF.fd # AAVMF
  pkgs.OVMF.fd
];
```

and not just ovmf.fd. The broken symlink for aavmf will probably still be there, but virt-manager won't complain about missing efi binaries and use the correct store path instead. Hope it helps :)

**Comment by robertosw** [comment]
> Just in case anyone else has the same problem, make sure to have
> 
> ```nix
> virtualisation.libvirtd.qemu.ovmf.packages = [
>   pkgs.pkgsCross.aarch64-multiplatform.OVMF.fd # AAVMF
>   pkgs.OVMF.fd
> ];
> ```
> 
> and not just ovmf.fd. The broken symlink for aavmf will probably still be there, but virt-manager won't complain about missing efi binaries and use the correct store path instead. Hope it helps :)

The AAVMF package also solved this issue for me, even though I already had `OVMFFull` installed. Thanks alot 👍🏼 

However this only solved my problems with qemu + virtmanager, but an aarch64 docker container, emulated by qemu, still does not work. Could this be related?

**Comment by jonhermansen** [comment]
Sorry to necro an old thread, but I've been struggling for a few days to get virt-manager to display the architecture options on VM creation (just like the screenshot in this bug). I have an aarch64 VM already, I just want to import it into libvirt. I tried adding some of the configs from this thread, to no avail. Does anyone still have a working setup that they could share their config from?

At one point, I think I had this working on NixOS, but now I don't know.

Thank you.
